client/app.jsx

```diff
@@ -46,24 +49,31 @@ export default class App extends React.Component {
   }
 
   renderPage() {
-    const {path} = this.state.route;
+    const { path } = this.state.route;
     if (path === '') {
-      return <h1>Testing</h1>
+      return <Home/>;
     }
     if (path === 'sign-in' || path === 'sign-up') {
-      return <Auth/>
+      return <Auth/>;
     }
-    return (<h1>Not Found</h1>)
+    if (path === 'profile') {
+      return <Profile/>;
+    }
+    return (<h1>Not Found</h1>);
   }
 
   render() {
     if (this.state.isAuthorizing) return null;
+    const token = window.localStorage.getItem('sudoku-token');
```

Comment:
It's generally not considered good practice to grab stuff from local storage in render like this. Usually it's harmless, but your render methods should be kept "fast" when possible and localStorage is a **synchronous** hard drive read, so people recommend not to do it (myself included). Render can run many times and it can really slow down with blocking i/o operations. Instead, take a look at the react-context exercise. There is a guard in App that blocks the render until it finishes checking localStorage. 

`isAuthorizing` is not checking if the user is logged in. `isAuthorizing` is a "flag" determining if the app finished checking localStorage for the token (in componentDidMount)